src/snowflake/snowpark/types.py

```diff
@@ -447,8 +449,8 @@ class _PandasType(DataType):
 class PandasSeriesType(_PandasType):
     """Pandas Series data type."""
 
-    def __init__(self, element_type: Optional[DataType]) -> None:
```

Comment:
This is a bit of risky change but it solves a LOT of type hint mismatch in `udf_utils.py`. We should review the consequence of this change closely when merging to `main`.